pkg/util/common/common.go

```diff
@@ -30,3 +35,80 @@ func (s StringSet) GetAll() []string {
 	}
 	return res
 }
+
+// StructToMap converts a struct to a map[string]interface{} based on `json` annotations defaulting to field names
+func StructToMap(obj interface{}) map[string]interface{} {
```

Comment:
Can this be with some existing package like `github.com/fatih/structs`?

Reply:
Usually rather not a big fan of pulling dependencies for small stuff, especially if used in a very specific use case.

Reply:
You have a point! I was approaching it from having to write tests for StructToMap. :)